rotkehlchen/chain/ethereum/manager.py

```diff
@@ -974,3 +974,34 @@ def get_blocknumber_by_time(self, ts: Timestamp, etherscan: bool = True) -> int:
             except RemoteError:
                 pass
         return self._get_blocknumber_by_time_from_subgraph(ts)
+
+    def get_basic_contract_info(self, address: ChecksumEthAddress) -> Dict[str, Any]:
+        """
+        Query a contract address and return basic information as:
+        - Decimals
+        - name
+        - symbol
+        if it is provided in the contract. This method may raise:
+        - BadFunctionCallOutput: If there is an error calling a bad address
+        """
+        # The only information we assume is decimals to be 18
+        info: Dict[str, Any] = {'decimals': 18}
+        request = ('decimals', 'symbol', 'name')
+
+        def query_info(method: str) -> Union[str, int, None]:
+            try:
+                return self.call_contract(
+                    contract_address=address,
+                    abi=ERC20TOKEN_ABI,
+                    method_name=method,
+                )
+            except (BlockchainQueryError, RemoteError):
+                return None
+
+        for prop in request:
```

Comment:
Since the multicall contract requires that all the calls succeeded, if one fails at this method... Would it be correct to call each method individually? Do you know of any approach that is better?

Reply:
Hmmm you have a point here. I expected it would return a null result in the non existing function but no... if function does not exist then you get a `RemoteErrror` due to transaction reverting. Tested with:

```python
    contracts_file = Path(__file__).resolve().parent / 'contracts.json'
    with contracts_file.open('r') as f:
        contracts = json.loads(f.read())
    univ2factory = EthereumContract(
        address=contracts['UNISWAPV2FACTORY']['address'],
        abi=contracts['UNISWAPV2FACTORY']['abi'],
        deployed_block=0,  # whatever
    )

    ethereum = init_ethereum(
        rpc_endpoint=args.eth_rpc_endpoint,
        use_other_nodes=args.use_other_nodes,
    )
    calls = [(
        univ2factory.address,
        univ2factory.encode(method_name='allPairsLength', arguments=[]),
    ), (
        univ2factory.address,
        univ2factory.encode(method_name='allPairs', arguments=[7]),
    )
             ,(
        univ2factory.address,
        univ2factory.encode(method_name='notexisting', arguments=[]),
    )
             ]
    result = multicall(
        ethereum=ethereum,
        calls=calls,
        call_order=None
    )
    decoded_result = [
        univ2factory.decode(result[0], 'allPairsLength'),
        univ2factory.decode(result[1], 'allPairs', [1]),
        univ2factory.decode(result[2], 'notexisting'),
    ]
```

Works fine if you don't use the `notexisting` function. Fails with `RemoteError` if you do.

Okay so that's a good question on the approach here. Hmmmmm :thinking: 

Since in most cases all tokens should have all 3 functions implemented I am wondering if it would not be wise to still use the aggregation by default. And if it fails, then fallback to the 3 calls.

This way I believe our average case would have 1 RPC call, but worst case would be 4. Does that make sense?

Reply:
Yes that's what I thought and already implemented this solution. As you mention the average case will have 1 call and I expect this to be the most common. Thank you!